modules/localization/msf/common/util/time_conversion.h

```diff
@@ -0,0 +1,105 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+#pragma once
+
+namespace apollo {
+namespace localization {
+namespace msf {
+
+// array_size(a) returns the number of elements in a.
+template <class T, size_t N>
+constexpr size_t ArraySize(T (&)[N]) {
+  return N;
+}
+
+// Leap seconds change every a few years. See below for details.
+// http://www.leapsecond.com/java/gpsclock.htm
+// http://maia.usno.navy.mil/ser7/tai-utc.dat
+//
+// UNIX time counts seconds since 1970-1-1, without leap seconds.
+// GPS time counts seconds since 1980-1-6, with leap seconds.
+// When a leap second is inserted, UNIX time is ambiguous, as shown below.
+//    UNIX date and time      UNIX epoch     GPS epoch
+//    2008-12-31 23:59:59.0   1230767999.0   914803213.0
+//    2008-12-31 23:59:59.5   1230767999.5   914803213.5
+//    2008-12-31 23:59:60.0   1230768000.0   914803214.0
+//    2008-12-31 23:59:60.5   1230768000.5   914803214.5
+//    2009-01-01 00:00:00.0   1230768000.0   914803215.0
+//    2009-01-01 00:00:00.5   1230768000.5   914803215.5
+
+// A table of when a leap second is inserted and cumulative leap seconds.
+static constexpr int32_t LEAP_SECONDS[][2] = {
```

Comment:
GNSS DRIVER also have a file [`time_conversion.h`](https://github.com/ApolloAuto/apollo/blob/v3.1_dev/modules/drivers/gnss/util/time_conversion.h ) Why not combine these abstract methods of the same function into the time_util.h file?